api/src/pcapi/routes/backoffice_v3/collective_offer_templates.py

```diff
@@ -223,39 +191,148 @@ def get_reject_collective_offer_template_form(collective_offer_template_id: int)
+
+def _batch_validate_or_reject_collective_offer_templates(
+    validation: OfferValidationStatus, collective_offer_template_ids: list[int]
+) -> bool:
+    collective_offer_templates = educational_models.CollectiveOfferTemplate.query.filter(
+        educational_models.CollectiveOfferTemplate.id.in_(collective_offer_template_ids),
+        educational_models.CollectiveOfferTemplate.validation == OfferValidationStatus.PENDING,
+    ).all()
+
+    if len(collective_offer_template_ids) != len(collective_offer_templates):
+        flash(
+            "Seules les offres collectives vitrines en attente peuvent être validées"
+            if validation is OfferValidationStatus.APPROVED
+            else "Seules les offres collectives vitrines en attente peuvent être rejetées",
+            "danger",
+        )
+        return False
+
+    collective_offer_template_update_succeed_ids: list[int] = []
+    collective_offer_template_update_failed_ids: list[int] = []
+
+    for collective_offer_template in collective_offer_templates:
+        new_validation_status = validation
+        collective_offer_template.validation = new_validation_status
+        collective_offer_template.lastValidationDate = datetime.datetime.utcnow()
+        collective_offer_template.lastValidationType = OfferValidationType.MANUAL
+        if validation is OfferValidationStatus.APPROVED:
+            collective_offer_template.isActive = True
+
+        try:
+            db.session.commit()
+        except Exception:  # pylint: disable=broad-except
+            collective_offer_template_update_failed_ids.append(collective_offer_template.id)
+            continue
+
+        collective_offer_template_update_succeed_ids.append(collective_offer_template.id)
+
+        recipients = (
+            [collective_offer_template.venue.bookingEmail]
+            if collective_offer_template.venue.bookingEmail
+            else [recipient.user.email for recipient in collective_offer_template.venue.managingOfferer.UserOfferers]
+        )
+
+        transactional_mails.send_offer_validation_status_update_email(
+            collective_offer_template, new_validation_status, recipients
+        )
+
+    search.async_index_collective_offer_template_ids(collective_offer_template_update_succeed_ids)
+
+    if len(collective_offer_template_ids) == 1:
+        flash(
+            "L'offre collective vitrine a bien été validée"
+            if validation is OfferValidationStatus.APPROVED
+            else "L'offre collective vitrine a bien été rejetée",
+            "success",
+        )
+    else:
+        flash(
+            f"Les offres collectives vitrines {', '.join(map(str, collective_offer_template_update_succeed_ids))} ont bien été validées"
+            if validation is OfferValidationStatus.APPROVED
+            else f"Les offres collectives vitrines {', '.join(map(str, collective_offer_template_ids))} ont bien été rejetées",
```

Comment:
Je suppose que par cohérence ça devrait être :
```suggestion
            else f"Les offres collectives vitrines {', '.join(map(str, collective_offer_template_update_succeed_ids))} ont bien été rejetées",
```